Orm/Xtensive.Orm/Orm/Session.Persist.cs

```diff
@@ -18,6 +18,8 @@ namespace Xtensive.Orm
 {
   public partial class Session
   {
+    private static readonly IDisposable EmptyDisposable = new Disposable(b => { return; });
```

Comment:
This idea of having one object which is useless in the most cases bugs me.

What if we used null-object pattern for pinner like this
```csharp
pinner = configuration.Supports(SessionOptions.AutoSaveChanges)
   ? new Pinner(this)
   : new NullPinner();
```
?

We would move ```EmptyDisposable``` to ```NullPinner``` and make ```NullPinner.RegisterRoot``` method return this cached object. The ```EmptyDisposable``` would be initialized only when it might be used.

It would allow us to have changes disabled like so

```csharp
public IDisposable DisableSaveChanges(IEntity target)
{
  EnsureNotDisposed();
  ArgumentValidator.EnsureArgumentNotNull(target, "target");
  var targetEntity = (Entity) target;
  targetEntity.EnsureNotRemoved();
  return pinner.RegisterRoot(targetEntity.State);
}
```

The second DisableSaveChanges would look like

```csharp
public IDisposable DisableSaveChanges()
{
  if (!Configuration.Supports(SessionOptions.AutoSaveChanges)) {
    return NullPinner.EmptyDisposable; // No need to pin in this case
  }
  if (disableAutoSaveChanges)
    return null;

  disableAutoSaveChanges = true;
  return new Disposable(_ => {
    disableAutoSaveChanges = false;
    InvalidateEntitySetsWithInvalidState();
  });
}
```

How do you feel about that?

Reply:
We can do even simpler:
return `null` instead of `EmptyDisposable`.
`null` is legal input for `using` statement.

Reply:
I suppose we can. Just don't forget to change methods' summaries to not mislead the user.

Reply:
Done.
The `<summary>` docs are already correct about returning `null`

Reply:
Well, you did forget the summaries :) Let me help you with this, lets commit following...

```csharp
/// <summary>
/// Temporarily disables all save changes operations (both explicit ant automatic) 
/// for specified <paramref name="target"/>.
/// Such entity is prevented from being persisted to the database,
/// when <see cref="SaveChanges"/> is called or query is executed.
/// If persist is to be performed due to starting a nested transaction or committing a transaction,
/// the presence of such an entity will lead to failure.
/// If <paramref name="target"/> is not present in the database,
/// all entities that reference <paramref name="target"/> are also pinned automatically.
/// </summary>
/// <param name="target">The entity to disable persisting.</param>
/// <returns>
/// A special object that controls lifetime of such behavior if <paramref name="target"/> was not previously processed by the method
/// and automatic saving of changes is enabled (<see cref="SessionOptions.AutoSaveChanges"/>),
/// otherwise <see langword="null"/>.
/// </returns>
public IDisposable DisableSaveChanges(IEntity target)
{
  EnsureNotDisposed();
  ArgumentValidator.EnsureArgumentNotNull(target, "target");
  if (!Configuration.Supports(SessionOptions.AutoSaveChanges))
    return null; // No need to pin in this case

  var targetEntity = (Entity) target;
  targetEntity.EnsureNotRemoved();
  return pinner.RegisterRoot(targetEntity.State);
}

/// <summary>
/// Temporarily disables only automatic save changes operations before queries, etc.
/// Explicit call of <see cref="SaveChanges"/> will lead to flush changes anyway.
/// If save changes is to be performed due to starting a nested transaction or committing a transaction,
/// active disabling save changes scope will lead to failure.
/// <returns>A special object that controls lifetime of such behavior if there is no active scope
/// and automatic saving of changes is enabled (<see cref="SessionOptions.AutoSaveChanges"/>),
/// otherwise <see langword="null"/>.</returns>
/// </summary>
public IDisposable DisableSaveChanges()
{
  if (!Configuration.Supports(SessionOptions.AutoSaveChanges) || disableAutoSaveChanges) {
    return null; // No need to pin in these cases
  }

  disableAutoSaveChanges = true;
  return new Disposable(_ => {
    disableAutoSaveChanges = false;
    InvalidateEntitySetsWithInvalidState();
  });
}
```

Reply:
Done
Thank you!